packages/cli/src/ui/components/SettingsDialog.test.tsx

```diff
@@ -928,6 +928,224 @@ describe('SettingsDialog', () => {
     });
   });
 
+  describe('Race Condition Regression Tests', () => {
+    it('should not reset sibling settings when toggling a nested setting multiple times', async () => {
+      vi.mocked(saveModifiedSettings).mockClear();
+
+      const FAKE_SCHEMA: SettingsSchemaType = {
```

Comment:
nit: move the fake schema outside of the test case to the top of the file  as it is duplicated in these two tests.